cmd/clusterctl/cmd/topology.go

```diff
@@ -0,0 +1,311 @@
+	}
+}
+
+func runTopologyDryRun() error {
+	c, err := client.New(cfgFile)
+	if err != nil {
+		return err
+	}
+
+	raw, err := os.ReadFile(dr.file)
+	if err != nil {
+		return errors.Wrap(err, "failed to read input file")
+	}
+
+	objs, err := utilyaml.ToUnstructured(raw)
+	if err != nil {
+		return errors.Wrap(err, "failed to convert file to list of objects")
+	}
+
+	out, err := c.DryRunTopology(client.DryRunOptions{
+		Kubeconfig: client.Kubeconfig{Path: dr.kubeconfig, Context: dr.kubeconfigContext},
+		Objs:       convertToPtrSlice(objs),
+		Cluster:    dr.cluster,
+	})
+	if err != nil {
+		return err
+	}
+	return printDryRunOutput(out, dr.outDir)
+}
+
+func printDryRunOutput(out *cluster.DryRunOutput, outdir string) error {
+	printAffectedClusterClasses(out)
+	printAffectedClusters(out)
+	if len(out.Clusters) == 0 {
+		// No affected clusters. Return early.
+		return nil
+	}
+	if out.ReconciledCluster.Name == "" {
+		fmt.Printf("No target cluster identified. Use --cluster to specify a target cluster to get detailed changes.")
+	} else {
+		printChangeSummary(out)
+		if err := writeOutputFiles(out, outdir); err != nil {
+			return errors.Wrap(err, "failed to write output files of target cluster changes")
+		}
+	}
+	fmt.Printf("\n")
+	return nil
+}
+
+func printAffectedClusterClasses(out *cluster.DryRunOutput) {
+	if len(out.ClusterClasses) == 0 {
+		// If there are no affected ClusterClasses return early. Nothing more to do here.
+		fmt.Printf("No ClusterClasses will be affected by the changes.\n")
+		return
+	}
+	fmt.Printf("The following ClusterClasses will be affected by the changes:\n")
+	for _, cc := range out.ClusterClasses {
+		fmt.Printf(" ＊ %s/%s\n", cc.Namespace, cc.Name)
+	}
+	fmt.Printf("\n")
+}
+
+func printAffectedClusters(out *cluster.DryRunOutput) {
+	if len(out.Clusters) == 0 {
+		// if there are not affected Clusters return early. Nothing more to do here.
+		fmt.Printf("No Clusters will be affected by the changes.\n")
+		return
+	}
+	fmt.Printf("The following Clusters will be affected by the changes:\n")
+	for _, cluster := range out.Clusters {
+		fmt.Printf(" ＊ %s/%s\n", cluster.Namespace, cluster.Name)
+	}
+	fmt.Printf("\n")
+}
+
+func printChangeSummary(out *cluster.DryRunOutput) {
+	if len(out.Created) == 0 && len(out.Modified) == 0 && len(out.Deleted) == 0 {
+		fmt.Printf("No changes detected for Cluster %q.\n", fmt.Sprintf("%s/%s", out.ReconciledCluster.Namespace, out.ReconciledCluster.Name))
+		return
+	}
+
+	fmt.Printf("Changes for Cluster %q: \n", fmt.Sprintf("%s/%s", out.ReconciledCluster.Namespace, out.ReconciledCluster.Name))
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Namespace", "Kind", "Name", "Action"})
+	table.SetCenterSeparator("")
+	table.SetColumnSeparator("")
+	table.SetRowSeparator("")
+	table.SetHeaderLine(false)
+	table.SetBorder(false)
+
+	// Add the created rows.
+	sort.Slice(out.Created, func(i, j int) bool { return lessByKindAndName(out.Created[i], out.Created[j]) })
```

Comment:
fyi. You can put the whole func into lessByKindAndName (including `func(i, j int)...`)